src/lib/y2storage/guided_proposal.rb

```diff
@@ -227,11 +268,11 @@ def populated_settings
     # to smaller disk size.
     #
     # @return [Array<String>] names of the chosen devices
-    def candidate_roots
-      return [populated_settings.root_device] if populated_settings.root_device
+    def candidate_roots_for(devices)
```

Comment:
What happens when the settings has one `root_device` and we are trying a proposal where devices has only one disk, but that disk is not the root device indicated in the settings? I guess no proposal is possible, isn't it?

Reply:
I guess the old method `candidate_roots` would work perfectly if we use `populated_settings` as it was doing. So, maybe it is not necessary to re-implement it as `candidate_roots_for(devices)`, or am I missing something?

Reply:
No, you are not missing anything :) Once again, you are right and that approach probably comes from my initial approach.

Thank you so much.